src/components/spreadsheet/utils/equipment-table-utils.ts

```diff
@@ -80,6 +81,93 @@ export const updateGeneratorCells = (params: CellEditingStoppedEvent) => {
     }
 };
 
+export const updateTwtCells = (params: CellEditingStoppedEvent) => {
+    const rowNode = params.node;
+    const colId = params.column.getColId();
+    const ratioRegulationTypeText =
+        params.data?.ratioTapChanger?.regulationType;
+    const phaseRegulationTypeText =
+        params.data?.phaseTapChanger?.regulationType;
+    const previousData = params.context.dataToModify;
+    switch (colId) {
+        case 'ratioTapChanger.regulationType':
+            if (ratioRegulationTypeText === REGULATION_TYPES.DISTANT.id) {
+                // This solve the problem of required on RatioRegulatingTerminal when we set regulationType to distant
+                const regulatingTerminalGenerator =
```

Comment:
I think that we should empty the "regulated side" when  the "regulation type" is set to "DISTANT"

Reply:
Maybe we should do the same for "regulated terminal" when the "regulation type" is set to "LOCAL"